sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -420,22 +426,24 @@ <T> CosmosPagedFlux<T> readAllItems(CosmosQueryRequestOptions options, Class<T>
         return UtilBridgeInternal.createCosmosPagedFlux(pagedFluxOptions -> {
             CosmosAsyncClient client = this.getDatabase().getClient();
             CosmosQueryRequestOptions nonNullOptions = options != null ? options : new CosmosQueryRequestOptions();
-            CosmosQueryRequestOptions requestOptions = clientAccessor.isClientTelemetryMetricsEnabled(client) ?
+            CosmosQueryRequestOptions requestOptions = clientAccessor.shouldEnableEmptyPageDiagnostics(client) ?
                 queryOptionsAccessor.withEmptyPageDiagnosticsEnabled(nonNullOptions, true)
                 : nonNullOptions;
+
+            queryOptionsAccessor.applyMaxItemCount(options, pagedFluxOptions);
```

Comment:
No - it just captures the MaxItemCount defined in the query request options in the flux options - because those are what is passed through the reactor pipeline and can be used to determine the defined max item count in diagnostics - doesn't change the behavior. This is the implementation of applyMaxItemCount

public void applyMaxItemCount(
                    CosmosQueryRequestOptions requestOptions,
                    CosmosPagedFluxOptions fluxOptions) {

                    if (requestOptions == null || requestOptions.getMaxItemCount() == null || fluxOptions == null) {
                        return;
                    }

                    if (fluxOptions.getMaxItemCount() != null) {
                        return;
                    }

                    fluxOptions.setMaxItemCount(requestOptions.getMaxItemCount());
                }